src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1036,43 +1033,43 @@ private TokenValidationResult ValidateToken(string token, JsonWebToken outerToke
                    && (tokenValidationResult.Exception as SecurityTokenUnableToValidateException).ValidationFailure != ValidationFailure.InvalidLifetime)
                    || tokenValidationResult.Exception.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))
                 {
```

Comment:
I commented in the other PR:

var resultType = tokenValidationResult.Exception.GetType();
if (resultType.Equals(typeof(SecurityTokenInvalidSignatureException))
   || resultType.Equals(typeof(SecurityTokenInvalidSigningKeyException))
   || resultType.Equals(typeof(SecurityTokenInvalidIssuerException))
   || (resultType.Equals(typeof(SecurityTokenUnableToValidateException))
   // we should not try to revalidate with the LKG or request a refresh if the token has an invalid lifetime
   && (tokenValidationResult.Exception as SecurityTokenUnableToValidateException).ValidationFailure != ValidationFailure.InvalidLifetime)
   || resultType.Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))

Reply:
Will this ever result in a null ref?
> (tokenValidationResult.Exception as SecurityTokenUnableToValidateException).ValidationFailure !

Reply:
Really good suggestion Danny; helps clean up the code a lot.
It shouldn't null ref; I check the exception type before casting it.